crates/sparrow-core/src/read_config.rs

```diff
@@ -0,0 +1,24 @@
+use arrow::datatypes::SchemaRef;
+
+/// Defines the configuration for a read from a source.
+#[derive(Debug)]
+pub struct ReadConfig {
+    /// The schema of the requested data.
+    ///
+    /// The actual schema of the input may have additional columns.
+    pub projected_schema: SchemaRef,
```

Comment:
This may not be plausible to have here. This may be just the `prepared_schema`. We'll see how it turns out

Reply:
I don't think it's needed here. Specifically, the logical plan has a `DataType` for what the read produces, which allows it to have the projected `DataType::Struct(projected_fields)`, so we can get that to pass into the read method when needed.